### ISSUE TITLE
Equivalent of Doxygen's option to ignore undocumented entities

### ISSUE THREAD

**Issue by anarthal** [cppalliance/mrdocs#758]
Doxygen has

```
HIDE_UNDOC_MEMBERS     = YES
HIDE_UNDOC_CLASSES     = YES
```

I could use an option to disable the generation of undocumented classes and members. For example, this is useful for template specializations.

**Comment by anarthal** [comment]
Yes, that's a real option (and IMO it should be the default), as something that's not documented should IMO not be used, regardless of where it's defined.

**Comment by alandefreitas** [comment]
I'm reopening this as the options we implemented in 7681b72 can filter documented symbols but cannot filter only members and classes.

```
HIDE_UNDOC_MEMBERS     = YES
HIDE_UNDOC_CLASSES     = YES
```

**Comment by anarthal** [comment]
In my case, I don't need the distinction. I don't know if people use one without the other, but I tend to set both of them.

**Comment by alandefreitas** [comment]
I'm having another look at the Doxygen documentation (as I'm implementing all other `EXTRACT_*` options in https://github.com/cppalliance/mrdocs/pull/858) and I'm not sure these `HIDE_*` options are a great idea for MrDocs. This issue describes the options as something to "disable the generation of undocumented classes and members," but I'm not sure that's exactly what they do.

- The extract options (`EXTRACT_*`) such as [EXTRACT_ALL](https://www.doxygen.nl/manual/config.html#cfg_extract_all) definitely disable the extraction and generation. 
- The hide options (`HIDE_*`) such as [HIDE_UNDOC_MEMBERS](https://www.doxygen.nl/manual/config.html#cfg_hide_undoc_members) seem to be more about how these undocumented symbols are presented in the documentation. 

Suppose it's about the presentation, considering how MrDocs renders class members. In that case, I don't see how it makes sense to extract the member and put it on that member list (the list only shows the name and brief, which won't exist either) without a link to a proper page that defines the synopsis. 

Or maybe it's not only about presentation. Maybe the wording in their documentation is confusing, and these `HIDE_*` options are just `EXTRACT_*` options in a confusing double negative. If that's the case, my questions are:

- Do you still think these options could be useful even though we already have all other extraction options (`extract-all`, `extract-*`, etc) and the warnings for undocumented members?

If so, I could create the same options but make the name clearer by using `extract-` instead of `hide-`, so the new options would be:

- `extract-undocumented-members`, 
- `extract-undocumented-classes`, 
- `extract-undocumented-namespaces`
- `extract-friend-compounds`

The other `HIDE_` options in doxygen (such as HIDE_SCOPE_NAMES) are completely unrelated to extraction.

**Comment by alandefreitas** [comment]
> One thing I find confusing is HIDE_UNDOC_MEMBERS "has no effect if [EXTRACT_ALL](https://www.doxygen.nl/manual/config.html#cfg_extract_all) is enabled". But it EXTRACT_ALL is disabled, aren't these undocumented members not going to be extracted anyway? That's why is seems to be about presentation.

The only way I'm making sense of this is if `EXTRACT_ALL`  being false still extracts everything unless it's also disabled by `HIDE_UNDOC_MEMBERS`, `HIDE_UNDOC_CLASSES`, or `HIDE_UNDOC_NAMESPACES`. Everything else (functions, variables, enums, etc) is still extracted, even if undocumented.

**Comment by alandefreitas** [comment]
> I don't see much value in replicating it as it is

I'm closing the issue as the options we have are enough then.
